### ISSUE TITLE
forward signal to child

### ISSUE THREAD

**Issue by wescande** [anishathalye/dotbot#184]
Hi !

I put a `chsh -s /bin/zsh` command in my dotfiles, and I send a Ctrl-C (SIGINT).
I received a pretty log ("Operation aborted"), but the SIGINT is not forward to chsh. Even if in a lot of situation, child does not care about parent status, here we have the following problem: termcap are not restored and the terminal is still in password input mode, which mean i have to kill the terminal and restart it (or reset but it's the same).

Thanks !

**Comment by anishathalye** [comment]
Hmm, I can't reproduce this behavior, in the sense that:

1. Use a `.install.conf.yaml` with a single line, `- shell: [chsh]`
1. Run `./install`
1. Press Ctrl-C

When I do this, it produces the following output:

```
chsh
Password for anish:

==> Operation aborted
```

And my terminal is no longer in password input mode.

Am I misunderstanding something?

**Comment by wescande** [comment]
Hi, Thanks for your reply

No you understand it right.

I did a fresh and simple yaml like your's, (and setting stdin + stdout + stderr to true)
This time _it works_.
As i was pretty sure not to be mad i start a fresh VM and try it. This time _it didn't work_.

By itterate, at came to the conclusion the issue is under **BASH** , not with **ZSH**.
Could you try it on bash ?

**Comment by anishathalye** [comment]
Hmm, I tried what I wrote above with bash as well, but on my regular macOS setup, and it gives the same results for me (doesn't get stuck in the password prompt).

(`bash --version` gives `GNU bash, version 5.0.0(1)-release (x86_64-apple-darwin18.2.0)`)

**Comment by wescande** [comment]
I tried on Linux mint (GNU bash, version 4.4.19(1)-release (x86_64-pc-linux-gnu))
And Manjaro (GNU bash, version 5.0.7(1)-release (x86_64-pc-linux-gnu)).
I tried with and without bashrc, I tried gnome terminal, Xterm, rxvt, Tmux.
I changed you're plugins/shell.py to proper handler signal and kill subprocess before exiting (so useless to do so but...)
In all case I reproduce the issue with bash not with zsh.
Now I will close this issue because I assume it's a problem with linux chsh + bash, not with Dotbot.
(I change chsh command with other command asking for password like getpass, even making custom one and it's works fine every time)

sorry to wasted your time
